internal/cmp/normalize.go

```diff
@@ -25,11 +24,11 @@ func SemanticEqual[T Normalizer[T]](this, that T) (bool, error) {
 	return reflect.DeepEqual(thisResult, thatResult), nil
 }
 
-func NormalizeSlice[S ~[]E, E any](slice S, cmp func(a, b E) int) S {
+func NormalizeSlice[S ~[]E, E any](slice S, less func(i, j int) bool) S {
```

Comment:
I made the `cmp` semantics consciously and would like to stay with that. The `less` semantics are deprecated in the Go SDK:
```
$ go doc sort.SliceStable
...
    Note: in many situations, the newer slices.SortStableFunc function is more
    ergonomic and runs faster.
```

Reply:
```
$ go doc slices.SortStableFunc
package slices // import "slices"

func SortStableFunc[S ~[]E, E any](x S, cmp func(a, b E) int)
    SortStableFunc sorts the slice x while keeping the original order of equal
    elements, using cmp to compare elements in the same way as SortFunc.
```

Reply:
ok, but the new semantics are pretty difficult for ordering several chained fields... maybe I can convert the fields to string and use a string comparison.

Reply:
@josvazg for `less`-semantics the `cmp` function has to return <0.

Reply:
yes, but also 0 is equal and >0 if greater. I will think this carefuly